src/shared/components/column-select/ColumnSelectDragDrop.tsx

```diff
@@ -20,55 +20,69 @@ const ColumnSelectDragDrop: FC<ColumnSelectDragDropProps> = ({
   columns,
   onDragDrop,
   onRemove,
-}) => (
-  <DragDropContext
-    onDragEnd={(result) => {
-      if (result.destination) {
-        onDragDrop(result.source.index, result.destination.index);
+}) => {
+  const previousColumns = useRef(columns);
+
+  useEffect(() => {
+    if (columns.length > previousColumns.current.length) {
+      const dndList = document.querySelector('.column-select-drag-drop__list');
+      if (dndList) {
+        dndList.scrollLeft = dndList.scrollWidth;
       }
-    }}
-  >
-    <Droppable droppableId="droppable" direction="horizontal">
-      {(droppableProvided) => (
-        <div
-          ref={droppableProvided.innerRef}
-          className={bem({
-            b: 'column-select-drag-drop',
-            e: 'list',
-          })}
-          {...droppableProvided.droppableProps}
-        >
-          {columns.map(({ itemId, label }, index) => (
-            <Draggable
-              key={itemId}
-              draggableId={itemId}
-              index={index}
-              disableInteractiveElementBlocking
-            >
-              {(draggableProvided, snapshot) => (
-                <div
-                  ref={draggableProvided.innerRef}
-                  {...draggableProvided.draggableProps}
-                  {...draggableProvided.dragHandleProps}
-                >
-                  <Chip
-                    disabled={false}
-                    onRemove={() => onRemove(itemId)}
-                    className={cn({
-                      'chip--dragging': snapshot.isDragging,
-                    })}
+    }
+    previousColumns.current = columns;
+  }, [columns]);
```

Comment:
Do you mind just commenting what this is accomplishing and the cases when this is important?